src/exec/command.rs

```diff
@@ -93,6 +93,41 @@ pub fn execute_commands<I: Iterator<Item = io::Result<Command>>>(
     ExitCode::Success
 }
 
+pub fn execute_commands_filtering<I: Iterator<Item = io::Result<Command>>>(
+    path: &std::path::Path,
+    cmds: I,
+    out_perm: &Mutex<()>,
+) -> ExitCode {
+    let mut output_buffer = OutputBuffer::new(out_perm);
```

Comment:
Writing to this output buffer skips all the logic in output.rs. 

The filter code shouldn't be handling output at all. 

Rather, this should be used as one of the filters that we use to exclude files to be processed  in walk.rs in the `spawn_senders` function.

Reply:
I based this on how the `--exec` command does the same. Should we change that as well?

Reply:
No, with exec, the output of the command itself is the desired output, but with filter, the desired output is the path (but possibly with some transformations done on it).

The question here is what should we do with the output from the filter command, if any (I imagine in most cases there won't be any).

Some options are:

1.  Pipe to /dev/null or equivalent 
2. Read, and immediately drop
3. Collect into buffer, then write to stderr, so that error messages are preserved.
4. Collect all filter output, then print all of it at the end (probably on stderr). 

I do think if we do print the output it should be to stderr.